### ISSUE TITLE
setting root-dir on remote systems by environment variables

### ISSUE THREAD

**Issue by ColemanTom** [metomi/rose#2162]
We have some paths which are set dynamically based on the user or some other control item. For example, a development user may have a path to store data in at `/a/b/foo_dev/dev` and a production user might have `/a/b/foo_op/op`. I do not see an easy way to share a rose-suite.conf file, using the `root-dir{...}` approach to setup share/work/home directories. For discussion, assume you can know that remote path at install time only as it might vary depending on various factors.

I believe like the only environment variables available in the root-dir setup are from what is available on ssh to the remote host and `CYLC_VERSION` and `ROSE_VERSION`, but it isn't something I have looked at in high levels of detail.

Ideally I think it would be useful to be able to do something like:
```
root-dir{share}=remote_machine=$MY_SHARE_PATH
```
and then the user setup MY_SHARE_PATH  and it makes its way to the install command via some mechanism which tells rose to pass that variable to the command along with the `*_VERSION` variables. Pre determined variable names could work perhaps (settable in a configuration file controlled at the rose installation level)?

Or, perhaps something like:
```
root-dir-pre-command="some_command_which_sets_up_variables_as_required"
root-dir{share}=remote_machine=$MY_PATH_VARIABLE
```
Then, part of the SSH command would be to call some script/function that sets up known needed environment variables. For example, like through https://opus.nci.org.au/display/Help/Environment+Modules. Leading to some command like this being executed.
```
ssh -oBatchMode=yes -oConnectTimeout=..... CYLC_VERSION=7.6.0\ some_command_which_sets_up_variables_as_required\ \&\&\ rose\ suite-run\ ....
```

I'm new to this side of rose, so if there are much better ways to approach what I'm describing above that already exist, it would be great to know too.

Thanks for any thought or help on this topic.

**Comment by matthewrmshin** [comment]
Rose's normal behaviour is to do something like `ssh [ssh-options] my-host bash -lc 'rose ...'`. This allows the environment to be set up normally for users (like a normal login). However, on sites where you are unable to call `bash -l` on SSH, you should still be able to use `lib/bash/rose_init_site` to set things up for you. (Obviously, you want to put in some intelligence there, so that you only call the set up logic when required.)

https://github.com/metomi/rose/blob/master/lib/bash/rose_init_site.example

**Comment by matthewrmshin** [comment]
@ColemanTom Please let us know if the `rose_init_site` solution works or not.

**Comment by sadielbartholomew** [comment]
I am looking to close down Issues that may no longer be relevant. @ColemanTom please can you let us know whether you managed to use ``rose_init_site`` to do as you wished here? If so, the feature you requested should not be necessary & we can close this. Thanks.

**Comment by oliver-sanders** [comment]
We have made some changes in the upcoming versions of Rose and Cylc (Rose 2 and Cylc 8):

* `rose suite-run` is no more.
  * Consequently the `root-dir` configuration has gone too.
* `cylc install` is now used to install workflows.
  * The `root-dir` equivalent for `cylc install` is the `[install][symlink dirs]` config.
  * This can be configured at the site or user level but *not* the workflow level.
  * (although a CLI override exists if you want to change it on a case by case basis).

This page gives a quick overview of the change: [cylc 8.0b2 docs](https://cylc.github.io/cylc-doc/8.0b2/html/7-to-8/major-changes/cylc-install.html)

Cylc makes use of login shells so any required variables can still be configured in user/site shell profile files.

Closing this issue as it has become outdated, if there is still an issue or missing functionality feel free to raise in `cylc-flow`.
